### PR TITLE
Implemented missing tags of GHPRB

### PR DESCRIPTION
**Pull request by erenatas**
Implemented the following tags that have been missing:
- displayBuildErrorsOnDownstreamBuilds
- blackListCommitAuthor
- GhprbCommentFile
- GhprbNoCommitStatus

Also updated tests

Signed-off-by: Eren ATAS <eatas.contractor@libertyglobal.com>

### CODE REVIEW

File: jenkins_jobs/modules/triggers.py [hunk 1]
```diff
@@ -1246,10 +1253,12 @@ def github_pull_request(registry, xml_parent, data):
         ("cron", "cron", ""),
         ("trigger-phrase", "triggerPhrase", ""),
         ("skip-build-phrase", "skipBuildPhrase", ""),
+        ("black-list-commit-author", "blackListCommitAuthor", None),
```
Comment (romanek-adam) [inline jenkins_jobs/modules/triggers.py]: what's the meaning of None here? how does it translate to XML?

File: jenkins_jobs/modules/triggers.py [hunk 2]
```diff
@@ -1246,10 +1253,12 @@ def github_pull_request(registry, xml_parent, data):
...
         ("only-trigger-phrase", "onlyTriggerPhrase", False),
         ("github-hooks", "useGitHubHooks", False),
         ("permit-all", "permitAll", False),
         ("auto-close-on-fail", "autoCloseFailedPullRequests", False),
+        ("display-build-errors-on-downstream-builds", "displayBuildErrorsOnDownstreamBuilds", None)
```
Comment (romanek-adam) [inline jenkins_jobs/modules/triggers.py]: shouldn't the default be False, according to the description in documentation?

File: jenkins_jobs/modules/triggers.py [hunk 3]
```diff
@@ -1330,9 +1339,12 @@ def github_pull_request(registry, xml_parent, data):
         str(data.get("cancel-builds-on-update", False)).lower() == "true"
     )
 
+    comment_file = data.get("comment-file", "")
+    no_commit_status = data.get("no-commit-status", False)
+
     # We want to have only one 'extensions' subelement, even if status
     # handling, comment handling and other extensions are enabled.
-    if requires_status or requires_job_comment or cancel_builds_on_update:
+    if requires_status or requires_job_comment or cancel_builds_on_update or comment_file:
```
Comment (romanek-adam) [inline jenkins_jobs/modules/triggers.py]: or no_commit_status ?

File: jenkins_jobs/modules/triggers.py [hunk 4]
```diff
@@ -1172,6 +1172,8 @@ def github_pull_request(registry, xml_parent, data):
         in the pull request will trigger a build (default false)
     :arg str skip-build-phrase: when filled, adding this phrase to
         the pull request title or body will not trigger a build (optional)
+    :arg str black-list-commit-author: When filled, pull request commits from this user(s)
+        will not trigger a build. (Note: this can be overridden by job-specific configuration.)
```
Comment (romanek-adam) [inline jenkins_jobs/modules/triggers.py]: I don't understand what the "Note" says

Comment (erenatas) [reply jenkins_jobs/modules/triggers.py]: I directly copied it over the help button on the Jenkins UI. Honestly, I don't understand it either. Should I delete it?

Comment (romanek-adam) [reply jenkins_jobs/modules/triggers.py]: I think so, I doubt it would be accepted into upstream.

**Comment by romanek-adam** [comment]
@erenatas please run `tox` on this one.

```
black....................................................................Failed
```

**Comment by erenatas** [comment]
> @erenatas please run `tox` on this one.
> 
> ```
> black....................................................................Failed
> ```

Can you check it now?

File: jenkins_jobs/modules/triggers.py [hunk 5]
```diff
@@ -1172,6 +1172,8 @@ def github_pull_request(registry, xml_parent, data):
         in the pull request will trigger a build (default false)
     :arg str skip-build-phrase: when filled, adding this phrase to
         the pull request title or body will not trigger a build (optional)
+    :arg str black-list-commit-author: When filled, pull request commits from this user(s) (optional)
+        will not trigger a build (optional)
```
Comment (romanek-adam) [inline jenkins_jobs/modules/triggers.py:1176]: (optional) twice?

**Comment by romanek-adam** [comment]
> Can you check it now?

tox passes.

File: jenkins_jobs/modules/triggers.py [hunk 6]
```diff
@@ -1172,6 +1172,8 @@ def github_pull_request(registry, xml_parent, data):
         in the pull request will trigger a build (default false)
     :arg str skip-build-phrase: when filled, adding this phrase to
         the pull request title or body will not trigger a build (optional)
+    :arg str black-list-commit-author: When filled, pull request commits from this user(s)
```
Comment (romanek-adam) [inline jenkins_jobs/modules/triggers.py]: The type should probably be "list" and the name of the key should rather become "black-list-commit-authors".

Sorry for being annoying, I'm just thinking about upstreaming.

Comment (erenatas) [reply jenkins_jobs/modules/triggers.py]: No of course not, thanks for noticing :)

**Review by romanek-adam — APPROVED** [review]

**Review by vdanaila — APPROVED** [review]

**Pull request merged**